parse_lib.py

```diff
@@ -95,8 +94,9 @@ def resolve_relative_resource_urls(html_string):
     imgs = html.find_all("img", src=True)
     equations = html.find_all("object", data=True)
     list(map(update_anchor_href, anchors))
```

Comment:
I think it's a tradeoff between being terse in the code and being explicit. I think that if there is a `map` call with an unused return value, it's indicative that the purpose of the `map` was to generate side effects. However, it would definitely be more explicit if it were brought out in a loop.

With respect to the core principles behind what `map`s are, though, I don't think side effects make a ton of sense (i.e. from a mathematical perspective). Whether or not it's deceptive in the context of Python is a something that I think depends on how it is used by the community-- since I don't have as much experience working in a variety of different Python projects, I'll defer to your judgement on this one!